internal/pkg/xds/xds_mcp.go

```diff
@@ -0,0 +1,202 @@
+	if err := downstream.Send(&discovery.DiscoveryResponse{
+		TypeUrl:     typeUrl,
+		VersionInfo: version,
+		Resources:   mcpResources,
+		ControlPlane: &envoycfgcorev3.ControlPlane{
+			Identifier: os.Getenv("POD_NAME"),
+		},
+		Nonce: version,
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// pushToSubscribers pushes MCP resources to active subscribers.
+func (adss *adsServer) pushToSubscribers() error {
+	mcpResources, err := makeMCPResources(numMCPResources)
+
+	if err != nil {
+		return fmt.Errorf("creating MCP resource: %w", err)
+	}
+
+	adss.subscribers.Range(func(key, value any) bool {
+		log.Print("Sending to subscriber ", fmt.Sprintf(subIDFmtStr, key.(uint64)))
+
+		if err = value.(*subscriber).stream.Send(&discovery.DiscoveryResponse{
+			TypeUrl:     "networking.istio.io/v1alpha3/ServiceEntry",
+			VersionInfo: strconv.FormatInt(time.Now().Unix(), 10),
+			Resources:   mcpResources,
+			ControlPlane: &envoycfgcorev3.ControlPlane{
+				Identifier: os.Getenv("POD_NAME"),
+			},
+		}); err != nil {
+			log.Print("Error sending MCP resources: ", err)
+			value.(*subscriber).closeStream()
+			adss.subscribers.Delete(key)
+		}
+
+		return true
+	})
+
+	return nil
+}
+
+// closeSubscribers closes all active subscriber streams.
+func (adss *adsServer) closeSubscribers() {
+	adss.subscribers.Range(func(key, value any) bool {
+		log.Print("Closing stream of subscriber ", fmt.Sprintf(subIDFmtStr, key.(uint64)))
+		value.(*subscriber).closeStream()
+		adss.subscribers.Delete(key)
+
+		return true
+	})
+}
+
+const numMCPResources = 100
+
+// makeMCPResources returns n Istio ServiceEntry objects serialized as protocol
+// buffer messages.
+func makeMCPResources(n int) ([]*anypb.Any, error) {
+	mcpResources := make([]*anypb.Any, 0, numMCPResources)
+	for i := 0; i < n; i++ {
+		mcpRes, err := makeMCPServiceEntry(i)
+		if err != nil {
+			return nil, fmt.Errorf("creating MCP resource: %w", err)
+		}
+		mcpResources = append(mcpResources, mcpRes)
+	}
+
+	return mcpResources, nil
+}
+
+// makeMCPServiceEntry returns an Istio ServiceEntry serialized as a protocol
+// buffer message.
+func makeMCPServiceEntry(idx int) (*anypb.Any, error) {
+	seSpec := &istionetv1alpha3.ServiceEntry{
+		Hosts:    []string{fmt.Sprintf("test%03d.toto.com", idx)},
+		Location: istionetv1alpha3.ServiceEntry_MESH_EXTERNAL,
+		Ports: []*istionetv1alpha3.ServicePort{{
+			Number:   443,
+			Name:     "https",
+			Protocol: "TLS",
+		}},
+		Resolution: istionetv1alpha3.ServiceEntry_STATIC,
+		Endpoints: []*istionetv1alpha3.WorkloadEntry{{
+			Address: "192.0.0.2",
+		}},
+	}
+
+	mcpResBody := &anypb.Any{}
+	if err := anypb.MarshalFrom(mcpResBody, seSpec, proto.MarshalOptions{}); err != nil {
+		return nil, fmt.Errorf("serializing ServiceEntry to protobuf message: %w", err)
+	}
+
+	mcpResTyped := &mcpv1alpha1.Resource{
+		Metadata: &mcpv1alpha1.Metadata{
+			Name: fmt.Sprintf("istio-system/mcp-example-%03d", idx),
```

Comment:
If we return fake data, could we return working SE? Then I would be able to install Istio with this controller and test provided resources.
```suggestion
func makeMCPServiceEntry(hostname string) (*anypb.Any, error) {
	seSpec := &istionetv1alpha3.ServiceEntry{
		Hosts:    []string{hostname},
		Location: istionetv1alpha3.ServiceEntry_MESH_EXTERNAL,
		Ports: []*istionetv1alpha3.ServicePort{{
			Number:   443,
			Name:     "https",
			Protocol: "TLS",
		}},
		Resolution: istionetv1alpha3.ServiceEntry_DNS,
	}

	mcpResBody := &anypb.Any{}
	if err := anypb.MarshalFrom(mcpResBody, seSpec, proto.MarshalOptions{}); err != nil {
		return nil, fmt.Errorf("serializing ServiceEntry to protobuf message: %w", err)
	}

	mcpResTyped := &mcpv1alpha1.Resource{
		Metadata: &mcpv1alpha1.Metadata{
			Name: fmt.Sprintf("istio-system/se-%s", hostname),
```